DataFormats/MuonReco/src/MuonSelectors.cc

```diff
@@ -179,96 +179,82 @@ float muon::segmentCompatibility(const reco::Muon& muon, reco::Muon::Arbitration
           // original "match weight" is currently reduced by at least attenuate_weight_regain, variing with an error function down to 0 if the track is
           // inside the chamber.
           // remark: the additional scale of 0.5 normalizes Err to run from 0 to 1 in y
-          station_weight[i - 1] = station_weight[i - 1] * attenuate_weight_regain * 0.5 *
-                                  (TMath::Erf(stations_w_track_at_boundary[i - 1] / 6.) + 1.);
+          station_weight[i - 1] = station_weight[i - 1] * attenuate_weight_regain * 0.5f *
+                                  (std::erf(stations_w_track_at_boundary[i - 1] / 6.f) + 1.f);
         } else if (station_has_segmentmatch[i - 1] <= 0 &&
-                   stations_w_track_at_boundary[i - 1] == 0.) {  // no segment match and track well inside chamber
+                   stations_w_track_at_boundary[i - 1] == 0.f) {  // no segment match and track well inside chamber
           // full penalization
-          station_weight[i - 1] = 0.;
+          station_weight[i - 1] = 0.f;
         }
       } else {  // always fully penalize tracks with no matching segment, whether the segment is close to the boundary or not.
         if (station_has_segmentmatch[i - 1] <= 0)
-          station_weight[i - 1] = 0.;
+          station_weight[i - 1] = 0.f;
       }
 
       // if track has matching segment, but the matching is not high quality, penalize
-      if (station_has_segmentmatch[i - 1] > 0 && 42 == 42) {
```

Comment:
I'm curious: what was the original purpose of the `&& 42 == 42` part in this condition?
Maybe @drkovalskyi or @slava77 can remember?

Reply:
I'm not familiar with this part of the code

Reply:
I'm not familiar with this part of the code

Reply:
just a guess,
this looks like a leftover of a pattern `&& true` vs `&& false` from a development stage of a large block of code.